src/mlpack/core/data/text_options.hpp

```diff
@@ -104,6 +117,25 @@ class TextOptions : public MatrixOptionsBase<TextOptions>
     return *this;
   }
 
+  void Combine(const TextOptions& other)
+  {
+    // Combine all boolean options.
+    hasHeaders =
+        DataOptionsBase<MatrixOptionsBase<TextOptions>>::CombineBooleanOption(
+        hasHeaders, other.hasHeaders, "HasHeaders()");
+    missingToNan =
+        DataOptionsBase<MatrixOptionsBase<TextOptions>>::CombineBooleanOption(
+        missingToNan, other.missingToNan, "MissingToNan()");
+    categorical =
+        DataOptionsBase<MatrixOptionsBase<TextOptions>>::CombineBooleanOption(
+        categorical, other.categorical, "Categorical()");
+    semicolon =
+        DataOptionsBase<MatrixOptionsBase<TextOptions>>::CombineBooleanOption(
+        semicolon, other.semicolon, "Semicolon()");
+
+    // TODO before PR: we don't combine 'headers' or 'datasetInfo'---should we?
```

Comment:
@shrit what do you think of this?  What should we do with `headers` and `datasetInfo` when we are combining two `TextOptions`?